### PR TITLE
Use GLib.MenuModel for context menu

### PR DESCRIPTION
**Pull request by jeremypw**
This is similar to #751 but based on latest master and more closely following current code to reduce diff (a little).  Code cleanup can be done in separate PR.

Also added custom "accel" menu attribute to the items.  This does not seem to be working with the current version - maybe it ill work in Gtk4?

### CODE REVIEW

File: src/Widgets/TerminalWidget.vala [hunk 1]
```diff
@@ -42,12 +42,13 @@ namespace Terminal {
             }
         }
 
-        private Gtk.Menu menu {
+        private Menu context_menu_model {
```
Comment (danirabbit) [inline src/Widgets/TerminalWidget.vala:45]: Hm so I get a warning here about:
```
../src/Widgets/TerminalWidget.vala:45.9-45.39: warning: Terminal.TerminalWidget.context_menu_model hides inherited property `Vte.Terminal.context_menu_model'. Use the `new' keyword if hiding was intentional
   45 |         private Menu context_menu_model {
```

I'm actually not sure what is the purpose of this property? it seems to only be used in a single private function

File: src/Widgets/TerminalWidget.vala [hunk 2]
```diff
@@ -42,12 +42,13 @@ namespace Terminal {
...
             get {
-                return main_window.menu;
+                return main_window.context_menu_model;
             }
         }
 
+        private Gtk.PopoverMenu context_popover;
```
Comment (danirabbit) [inline src/Widgets/TerminalWidget.vala:51]: This should probably not be `Gtk.PopoverMenu` in Gtk3, but instead `Gtk.Menu` and we can change it to `PopoverMenu` in Gtk4

File: src/Widgets/TerminalWidget.vala [hunk 3]
```diff
@@ -460,11 +451,26 @@ namespace Terminal {
...
+
+            // Popup context menu below cursor position
+            context_popover = new Gtk.PopoverMenu () {
+                relative_to = this,
+                pointing_to = rect
+            };
+
+            context_popover.bind_model (context_menu_model, null);
+            context_popover.show_all ();
+            context_popover.popup ();
```
Comment (danirabbit) [inline src/Widgets/TerminalWidget.vala:471]: I think `show_all` and `popup` are redundant here. I'm not sure `show_all` is necessary when using a menu from a model

**Review by danirabbit — CHANGES_REQUESTED** [review]

File: src/Widgets/TerminalWidget.vala (hunk 1 above)
Comment (jeremypw) [reply src/Widgets/TerminalWidget.vala:45]: Now removed.

File: src/MainWindow.vala [hunk 4]
```diff
@@ -154,53 +156,57 @@ namespace Terminal {
...
             primary_selection = Gtk.Clipboard.get (Gdk.Atom.intern ("PRIMARY", false));
 
-            var open_in_browser_menuitem = new Gtk.MenuItem () {
-                action_name = ACTION_PREFIX + ACTION_OPEN_IN_BROWSER
-            };
-            open_in_browser_menuitem_label = new Granite.AccelLabel.from_action_name (
-                "", open_in_browser_menuitem.action_name
+            //Window actions
+            open_in_browser_menuitem = new MenuItem (
+               "DUMMY open in name",
```
[the remark this answers is not in the document]
Comment (jeremypw) [reply src/MainWindow.vala:161]: Fixed

File: src/Widgets/TerminalWidget.vala (hunk 3 above)
Comment (jeremypw) [reply src/Widgets/TerminalWidget.vala:471]: It does not work in Gtk3 without these.  Revisit in Gtk4

**Comment by danirabbit** [comment]
@jeremypw proposed https://github.com/elementary/terminal/pull/770 to merge into this branch. Fixes missing accels

**Review by danirabbit — APPROVED** [review]

**Pull request merged**